### PR TITLE
Add secure ingress task

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**

### CODE REVIEW

File: content/docs/tasks/traffic-management/secure-ingress.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                prefix: /delay
+            route:
+            - destination:
+                port:
+                  number: 8000
+                host: httpbin
+        EOF
+    ```
+
+1.  Access the _httpbin_ service by HTTPS, sending an HTTPS request by _curl_ to `SECURE_INGRESS_PORT`.
```
Comment (frankbu) [inline content/docs/tasks/traffic-management/secure-ingress.md]: You deleted from the ingress task the explanation of what --resolve is for:

**Note the `--resolve` flag that we use this time. Unlike HTTP requests, the `Host` header that we pass will be encrypted this time, so the ingress gateway will not be able to use it to match the request to our configuration. The `--resolve` flag instructs _curl_ to supply the [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) value "httpbin.example.com" when accessing the gateway IP over TLS.**

It would be good to work something similar into this task.

Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress.md]: @frankbu Good catch, it was not my intention. I will put this part back, rewriting it to use "you" instead "we".

Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress.md]: @frankbu I put a comment about `--resolve` flag, please review.

File: content/docs/tasks/traffic-management/secure-ingress.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+## Configure a TLS ingress gateway
+
+In this subsection we configure an ingress gateway with the port 443 to handle HTTPS traffic. We create a secret
+with a certificate and a private key. Then we create a `Gateway` definition that contains a `server` on the port 443.
+
+1. Create a Kubernetes `Secret` to hold the server's certificate and private key. Use `kubectl` to create the secret
+`istio-ingressgateway-certs` in namespace `istio-system` . The Istio gateway will load the secret automatically.
+
+    > The secret MUST be called `istio-ingressgateway-certs` in the `istio-system` namespace, or it will not
```
[the remark this answers is not in the document]
Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress.md]: @geeknoid The issue is that this task involves several risky points, this is why there are so many notes. Each note relates to a possible user mistake that can cause Istio ingress to break. There are two notes about using a wrong name of a certificate, two notes about using a wrong location of a certificate, and two notes about waiting for the gateway definition propagation, otherwise the users could be confused by unexpected results. The task itself is a bit tricky.

File: content/docs/tasks/traffic-management/secure-ingress.md (hunk 1 above)
Comment (frankbu) [reply content/docs/tasks/traffic-management/secure-ingress.md]: I would reword this step a little, and break up the text like this:

1. Access the _httpbin_ service with HTTPS by sending an `https` request using _curl_ to `SECURE_INGRESS_PORT`. 

   The --resolve flag is used to instruct _curl_ to supply the [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) value "httpbin.example.com" when accessing the gateway IP over TLS. The `--cacert` option instructs _curl_ to use your generated certificate to verify the server. 

   By sending the request to the `/status/418` URL path, you get a nice visual clue that your _httpbin_ service was indeed accessed. The _httpbin_ service will return the [418 I'm a Teapot](https://tools.ietf.org/html/rfc7168#section-2.3.3) code.

File: content/docs/tasks/traffic-management/secure-ingress.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    ```command
+    $ generate.sh httpbin.example.com <password>
+    ```
+
+    The command will generate four directories: `1_root`, `2_intermediate`, `3_application` and `4_client` with client
+    and server certificates you will use.
+
+## Configure a TLS ingress gateway
+
+In this subsection we configure an ingress gateway with the port 443 to handle HTTPS traffic. We create a secret
```
Comment (frankbu) [inline content/docs/tasks/traffic-management/secure-ingress.md]: you configure

File: content/docs/tasks/traffic-management/secure-ingress.md (hunk 1 above)
Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress.md]: @frankbu Done, thanks.

**Comment by vadimeisenbergibm** [comment]
@frankbu @geeknoid Please note additional commit https://github.com/istio/istio.github.io/pull/1590/commits/91dc65d2bcb7232524cc3f1f7c151e889d7cc361 .

Please approve.

**Comment by geeknoid** [comment]
Hi, I did some refactoring and the files you edited has moved in the repo. Please rebase and push again. Sorry for the inconvenience.

**Comment by vadimeisenbergibm** [comment]
Hi @geeknoid, I have rebased, refactored and pushed, everything was cleanly merged, without any problems.

**Comment by vadimeisenbergibm** [comment]
@frankbu Moved the requirement for macOS _curl_ to the _Before you begin_ section, and put a one-sentence reminder about it in _Troubleshooting_.

File: content/docs/tasks/traffic-management/secure-ingress/index.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
+---
+title: Secure Ingress Traffic
```
Comment (rshriram) [inline content/docs/tasks/traffic-management/secure-ingress/index.md]: can we clearly name this as Securing Gateways with HTTPS?

Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress/index.md]: I would propose `Secure Ingress Gateways with TLS`, since we are talking about configuring  TLS/mTLS for ingress traffic. Or  `Secure Ingress Traffic with TLS`, `Configure TLS for Ingress Traffic` or `Configure TLS for Ingress Gateways`. More ideas are welcome.

@geeknoid @frankbu @rshriram

Comment (vadimeisenbergibm) [reply content/docs/tasks/traffic-management/secure-ingress/index.md]: @geeknoid @frankbu @rshriram could you please agree on the title? See https://github.com/istio/istio.github.io/pull/1590#discussion_r198693721 and https://github.com/istio/istio.github.io/pull/1590#discussion_r198668429.

The tests pass, I would like to merge the PR.

Comment (geeknoid) [reply content/docs/tasks/traffic-management/secure-ingress/index.md]: Sorry, I merged it already. I like Shriram's suggestion: Securing Gateways with HTTPS
